.github/workflows/check-migrations.yml

```diff
@@ -0,0 +1,59 @@
+name: Check Migrations
+
+on:
+  push:
+    branches: ["master"]
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-migrations:
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        runtime:
+          - astar
+          - shiden
+          - shibuya
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18.x
+
+      - name: Install deps
+        run: sudo apt -y install protobuf-compiler
+
+      - name: Install & display rust toolchain
+        run: rustup show
+
+      - name: Check targets are installed correctly
+        run: rustup target list --installed
+
+      - name: Build EXTRA_ARGS
+        if: ${{ !startsWith(github.ref, 'refs/tags/v') }}
```

Comment:
Is this needed since you already check it as the entry condition for the script to run?

Reply:
Yes it's needed. It shouldn't check spec version unless you're releasing. This will run on master or manually triggered to your branch

Reply:
Oh I see, thanks!